chase_step1a_OutcomeGroupsUsingCleanedEncounters.sas

```diff
@@ -272,154 +364,193 @@ DelivPR_Inp    DelivPR_Out    DelivPR_EM    Frequency     Percent     Frequency
+		/*Create some new variables in the outcomesbygroup_encgap dataset*/
+      	/** first subquery counts at the group level (1+ encounters)**/
+       (select distinct patient_deid, outcomegrp, min(startdt) as Dt_OutcomeGroup_Start, 
+				max(enddt) as Dt_OutcomeGroup_End, 
+             	count(distinct enc_key) as OutcomeGroup_Encounters_N,
+             	count(distinct enc_date) as OutcomeGroup_Dates_N, 
+             	count(distinct outcome) as OutcomeGroup_Outcomes_N ,  /*CDL: Change. Commented out the next line and retained this var*/
+/*             	count(distinct Case when dxprrx ne '001' then Outcome end ) as OutcomeGroup_Outcomes_N ,  */
+             	count(distinct Case when dxprrx not in ('000' '003' '002' '001') then Outcome end ) as OutcomeGroup_OutcomesNoRX_N ,  
+              	calculated dt_outcomegroup_end - calculated dt_outcomegroup_start +1 as OutcomeGroup_Span ,
 /*              COUNT(distinct case when DelivPR_Inp then enc_key end) as OutcomeGroup_InpPrEncounters_N ,*/
-              COUNT(distinct case when DelivPR_Inp then enc_key end) as OutcomeGroup_DelivPRInpatEnc_N ,
-              COUNT(distinct case when DelivPR_Out then enc_key end) as OutcomeGroup_DelivPROutpatEnc_N ,
-              COUNT(distinct case when DelivPR_Em then enc_key end) as OutcomeGroup_DelivPREmergEnc_N ,
-             max(days_prevgroup) as Days_PrevOutcomeGroup
-        from outcomesbyGroup_&encgap. group by patient_deid, outcomegrp ) As Grp
+              	COUNT(distinct case when DelivPR_Inp then enc_key end) as OutcomeGroup_DelivPRInpatEnc_N ,
+              	COUNT(distinct case when DelivPR_Out then enc_key end) as OutcomeGroup_DelivPROutpatEnc_N ,
+              	COUNT(distinct case when DelivPR_Em then enc_key end) as OutcomeGroup_DelivPREmergEnc_N ,
+             	max(days_prevgroup) as Days_PrevOutcomeGroup
+        from outcomesbyGroup_&encgap. 
+		group by patient_deid, outcomegrp ) As Grp
+
 
       Left join
 
+	  /*Summarize the Dx Pr and Rx information for each outcome within the pregnancy outcome groups*/
       /** 2nd subquery counts at the outcome w/in group (1+ rows for same outcometype combined)**/
        (select distinct patient_deid, outcomegrp, outcome, 
-             outcome,max(dx) as Dx, max(pr) as PR, max(rx) as Rx, 
-             cats(max(dx),max(pr),max(rx)) as DxPrRx
+             /*outcome,*/max(dx) as Dx, max(pr) as PR,  /*CDL: Commented out 2nd outcome*/
+			 /*CDL: MAX(RX) does not give the desired output. Specifically, if they had both on 
+			 different encounters (observed), then we would only see 2 not 3, as would be correct*/
+/*			max(rx) as Rx, */
+			case
+				when count(distinct (case when rx=0 then . else rx end)) > 1 then 3 else max(rx) end as Rx,
+/*             cats(max(dx),max(pr),max(rx)) as DxPrRx*/
+			 cats(max(dx), max(pr), case when count(distinct (case when rx=0 then . else rx end)) > 1 then 3 else max(rx) end) as DxPrRx

```

Comment:
max(r) was not getting the appropriate results. Changed it to a case_when function that does.